pkg/formats/sniffer.go

```diff
@@ -46,6 +46,49 @@ func (fs *Sniffer) SniffReader(f io.ReadSeeker) (Format, error) {
 			fmt.Printf("WARNING: could not seek to beginning of file: %v", err)
 		}
 	}()
+
+	type SpecVersionStruct struct {
+		BomFormat       string `json:"bomFormat"`
+		CDXSpecVersion  string `json:"specVersion"`
+		SPDXSpecVersion string `json:"spdxVersion"`
+	}
+
+	decoder := json.NewDecoder(f)
+
+	var specversionjson SpecVersionStruct
+	err := decoder.Decode(&specversionjson)
+	if err == nil {
+		if specversionjson.BomFormat == "CycloneDX" {
+			if specversionjson.CDXSpecVersion == "1.3" {
+				return CDX13JSON, nil
+			} else if specversionjson.CDXSpecVersion == "1.4" {
+				return CDX14JSON, nil
+			} else if specversionjson.CDXSpecVersion == "1.5" {
+				return CDX15JSON, nil
+			} else {
+				// JSON + BomFormat CycloneDX but specVersion not 1.3, 1.4, or 1.5
+				return "", fmt.Errorf("unknown SBOM format")
+			}
+		} else {
+			// JSON but not CycloneDX so assuming SPDX
+			if specversionjson.SPDXSpecVersion == "SPDX-2.2" {
+				return SPDX22JSON, nil
+			} else if specversionjson.SPDXSpecVersion == "SPDX-2.3" {
+				return SPDX23JSON, nil
+			} else {
+				// JSON + not CycloneDX but spdxVersion not SPDX-2.2 or SPDX-2.3
+				return "", fmt.Errorf("unknown SBOM format")
+			}
+		}
+	}
+
+	// not JSON.  Parse line-by-line with string hacks
+
+	_, err = f.Seek(0, 0)
+	if err != nil {
+		fmt.Printf("WARNING: could not seek to beginning of file: %v", err)
```

Comment:
Dont print to STDOUT in the library. In fact, I think we should error here as a failure to rewind the pointer is going to end up in an error parsing the SBOM anyway,
```suggestion
		return fmt.Errorf("seeking to the beginning of SBOM file: %w", err)
```

Reply:
yes, thank you.  Fixed.